src/entities/institute/institute.model.ts

```diff
@@ -0,0 +1,76 @@
+import * as DB from '@nestjs/mongoose' // { Prop, Schema, SchemaFactory }
+import * as GQL from '@nestjs/graphql' // { Field, ObjectType, ID }
+import mongoose from 'mongoose'
+import autopopulate from 'mongoose-autopopulate'
+import * as Utils from '../../utils/Model'
+
+// Pagination
+import { Paginated, PaginatedType, withCursor } from '../../utils/Pagination'
+
+// Entities
+import { School } from '../school/school.model'
+import { Subject, PaginatedSubjects } from '../subject/subject.model'
+import { User, PaginatedUsers } from '../user/user.model'
+
+@GQL.ObjectType()
+@DB.Schema({
+  timestamps: true,
+  // autoIndex: true
+})
+export class Institute extends Utils.BaseModel {
+  constructor(school?: School) {
+    super()
+    this.school = school
+  }
+
+  @GQL.Field(() => GQL.ID)
+  _id: mongoose.Types.ObjectId
+
+  @GQL.Field(() => String)
+  @DB.Prop()
+  name: string
+
+  @GQL.Field(() => String)
+  @DB.Prop()
+  shortName: string
+
+  @GQL.Field(() => User, { nullable: false })
+  @DB.Prop({
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'User',
+    autopopulate: true,
+  })
+  createdBy: mongoose.Types.ObjectId | User
+
+  // *
+  // Relations
+  // *
+
+  @GQL.Field(() => Number)
+  @DB.Prop({ type: Number, default: 0 })
+  followerCounter: number
+
+  @GQL.Field(() => School, { nullable: true })
+  @DB.Prop({
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'School',
+    autopopulate: true,
+  })
+  school: mongoose.Types.ObjectId | School
+
+  // *
+  // Connections
+  // *
+
+  @GQL.Field(() => PaginatedSubjects, { nullable: true })
+  subjectsConnection: PaginatedType<Subject>
```

Comment:
@frankmangone I am not sure about naming this relations with Connection, is not wrong but redundant since this info is in the schema itself, right?

Reply:
Okay, open issue #26 for this purpose